submissions/mxmgny/a-tiny-js-world/index.js

```diff
@@ -1,43 +1,47 @@
-let cat = {
-    appearance: "cat",
-    name: "Garfield",
+const cat = {
+    appearance: 'cat',
+    name: 'Garfield',
     legs: 4,
     hands: 0,
-    gender: "male",
-    shout: "I love lazagna!",
-    friends:['Selina', 'Jeremy']
+    gender: 'male',
+    shout: 'I love lazagna!',
 };
-let cat_woman = {
-    appearance:"cat-woman",
-    name:"Selina",
-    legs:2,
-    hands:2,
-    gender: "female",
-    shout:cat['shout'],
-    friends: [cat['name'],'Sansa']
+const catWoman = {
+    appearance: 'cat-woman',
+    name: 'Selina',
+    legs: 2,
+    hands: 2,
+    gender: 'female',
+    shout: cat['shout']
 };
-let dog = {
-    appearance:"dog",
-    name: "Jerry",
-    gender: "male",
+const dog = {
+    appearance: 'dog',
+    name: 'Jerry',
+    gender: 'male',
     legs: 4,
     hands: 0,
-    shout: "Woof!",
-    friends: ["Sansa","Snoop Dog", "Balto", "Goofy"]
+    shout: 'Woof!'
 };
-let human = {
-    appearance:"human",
-    name:"Sansa",
-    gender:"female",
-    legs:2,
-    hands:2,
-    shout:"Winter is coming",
-    friends:[dog['name'], "Robb", "Arya", "Bran", "Rickon"]
+const human = {
+    appearance: 'human',
+    name: 'Sansa',
+    gender: 'female',
+    legs: 2,
+    hands: 2,
+    shout: 'Winter is coming'
 };
 
-let arr = [cat, dog, human,cat_woman];
+human.friends = [ dog.name,'Rob', cat.name ];
+dog.friends = [ human.name, 'Snoop Dog' ];
+catWoman.friends = [ human.name, cat.name ];
+cat.friends = [ human.name, catWoman.name] ;
 
-arr.forEach(obj => {
-    let friends = obj['friends'].join(', ');
-    print(`Hi, my name is <strong>${obj['name']}</strong>. I am <em>${obj['gender']} ${obj['appearance']}</em>.I have ${obj['legs']} legs and ${obj['hands']} hands. My friends are ${friends}. And you know what? ${obj['shout']}`);
-})
+
+[ cat, dog, human,catWoman ].forEach(obj => {
+    print(  [
+        `Hi, my name is <strong>${obj['name']}</strong>.`,
+        `I am <em>${obj['gender']} ${obj['appearance']}</em>.`,
+        `I have ${obj['legs']} legs and ${obj['hands']} hands.`,
+        `My friends are ${obj['friends'].join(', ')}.`,
```

Comment:
Now, as we filled friends array with the existing objects - we need some changes to our pattern to correctly print a friend's names only.
Any ideas how we may rewrite `${obj['friends'].join(', ')}` in this case?

Reply:
Since I need to go through every friendObj in object and get its name, i used map function because it returns  an array  of names which I concatenate into single string of friends names.